expertise/service/routes.py

```diff
@@ -0,0 +1,458 @@
+        # For each profile directory, look at each job
+        for profile_name in profile_names:
+            job_ids = get_subdirs(os.path.join(root_dir, profile_name))
+            error_dir = os.path.join(root_dir, profile_name, 'err.log')
+
+            # If no score file is present, clean up dir and write to error log
+            for job_id in job_ids:
+                job_dir = os.path.join(root_dir, profile_name, job_id)
+                score_dir, _ = get_score_and_metadata_dir(job_dir)
+                if score_dir is None:
+                    with open(error_dir, 'a+') as f:
+                        f.write(f"{job_id},Interrupted before completed")
+                    shutil.rmtree(job_dir)
+
+@BLUEPRINT.route('/test')
+def test():
+    """Test endpoint."""
+    flask.current_app.logger.info('In test')
+    return 'OpenReview Expertise (test)'
+
+@BLUEPRINT.route('/expertise', methods=['POST'])
+def expertise():
+    """
+    Submit a job to create a dataset and execute an expertise model based on the submitted configuration
+
+    :param token: Authorization from a logged in user, which defines the set of accessible data
+    :type token: str
+
+    :param name: A name describing the job being submitted
+    :type name: str
+
+    :param match_group: A group whose profiles will be used to compute expertise
+    :type match_group: str
+
+    :param paper_invitation: An invitation containing submissions used to compute expertise
+    :type paper_invitation: str
+    """
+    result = {}
+
+    token = flask.request.headers.get('Authorization')
+    in_test_mode = 'IN_TEST' in flask.current_app.config.keys()
+    
+    if not token and not in_test_mode:
+        flask.current_app.logger.error('No Authorization token in headers')
+        result['error'] = 'No Authorization token in headers'
+        return flask.jsonify(result), 400
+    
+    try:
+        user_config = flask.request.json
+        flask.current_app.logger.info('Received expertise request')
+        
+        if not in_test_mode:
+            openreview_client = openreview.Client(
+                token=token,
+                baseurl=flask.current_app.config['OPENREVIEW_BASEURL']
+            )
+            user_config['token'] = token
+            user_config['baseurl'] = flask.current_app.config['OPENREVIEW_BASEURL']
+        else:
+            openreview_client = mock_client()
+
+        profile, _ = get_profile_and_id(openreview_client)
+        job_id = enqueue_expertise(user_config, profile.id)
+
+        result['job_id'] = job_id
+        flask.current_app.logger.info('Returning from request')
+        
+    except openreview.OpenReviewException as error_handle:
+        flask.current_app.logger.error(str(error_handle))
+
+        error_type = str(error_handle)
+        status = 500
+
+        if 'not found' in error_type.lower():
+            status = 404
+        elif 'forbidden' in error_type.lower():
+            status = 403
```

Comment:
The OpenReview exceptions don't have the status value?

Reply:
Status here refers to the HTTP status code to be returned, not necessarily trying to find the status value of the OpenReview exception:
https://github.com/openreview/openreview-expertise/blob/fa8c9ce32653858df385f2d1e66baae9e71418da/expertise/service/routes.py#L244